### PR TITLE
Three tasks: upgrade and use latest gradle plugin features, improve DockerFile performance, wrap runtime exceptions

### PR DESCRIPTION
**Pull request by dmgaldi**
## Overview
* Upgrade user service to use gradle plugin for RAML generation
* Upgrade container core version to the latest
* Refine Dockerfile task ordering to improve cache hit %
* (Java code) Add exception wrappers around service domain logic to hide internals on 500s

Note: the makefile is now unnecessary, but is often convenient for development (no longer used in Dockerfile)

## Testing
* Ran gradle tasks successfully
* Created docker image successfully

### CODE REVIEW

**Comment by dmgaldi** [comment]
> I'm not actually sure I like this way better. Harder to track what all is going on. Like if I run make gen-jaxrs, I can track each step along the way. But by tacking on pre- and post-conditions to a "well known" gradle target, it could be a total mystery how it happens to someone not in the know.

Do you think we should have a gradle plugin task for fetching the RAML? I was hesitant since it's only done in EDA at the moment. I'm not sure how one would "opt into", "opt out of" or configure the common RAML fetch task if we put it in the plugin.

**Comment by dmgaldi** [comment]
> > I'm not actually sure I like this way better. Harder to track what all is going on. Like if I run make gen-jaxrs, I can track each step along the way. But by tacking on pre- and post-conditions to a "well known" gradle target, it could be a total mystery how it happens to someone not in the know.
> 
> Do you think we should have a gradle plugin task for fetching the RAML? I was hesitant since it's only done in EDA at the moment. I'm not sure how one would "opt into", "opt out of" or configure the common RAML fetch task if we put it in the plugin.

Talked about this after scrum, updated based on our conversation and Ellie's help

**Comment by ryanrdoherty** [comment]
@dmgaldi @Foxcapades Can you take a look at this as a final cut?  I think it's ready to merge; can maybe be a template to improve all the other services (esp EDA services).

**Comment by ryanrdoherty** [comment]
@steve-fischer-200 Please review the Java code

File: src/main/java/org/veupathdb/service/eda/us/model/UserDataFactory.java [hunk 1]
```diff
@@ -106,6 +111,29 @@ private String addSchema(String sqlConstant) {
     return sqlConstant.replace(SCHEMA_MACRO, _userSchema);
   }
 
+  /***************************************************************************************
+   *** Exception handling so underlying message is not exposed to user
```
Comment (steve-fischer-200) [inline src/main/java/org/veupathdb/service/eda/us/model/UserDataFactory.java:115]: These methods should be in FgpUtil

File: src/main/java/org/veupathdb/service/eda/us/model/UserDataFactory.java [hunk 2]
```diff
@@ -106,6 +111,29 @@ private String addSchema(String sqlConstant) {
...
+   **************************************************************************************/
+
+  private static final Function<Exception,RuntimeException> EXCEPTION_MAPPER = e -> {
+    Throwable root = e instanceof SQLRunnerException ? e.getCause() : e;
```
Comment (steve-fischer-200) [inline src/main/java/org/veupathdb/service/eda/us/model/UserDataFactory.java:119]: i don't fully understand why getCause() is what we want here

**Review by steve-fischer-200 — CHANGES_REQUESTED** [review]
looks good.  i requested changes cuz of my suggestion about moving methods to fgputil

**Pull request merged**